packages/react-components/src/components/gauge/utils/getThresholds.ts

```diff
@@ -1,29 +1,113 @@
-import { getDecimalValueFromMinMax } from './getDecimalValueFromMinMax';
-import { GaugeProps } from '../types';
+import { getThresholdRangeFromMinMax } from './getThresholdRangeFromMinMax';
+import { GaugeProps, GaugeSettings } from '../types';
 import { DEFAULT_GAUGE_PROGRESS_COLOR } from '../constants';
+import { ComparisonOperator } from '@iot-app-kit/core';
 
+type ThresholdRange = {
+  start?: number;
+  end?: number;
+  color: string;
+  comparisonOperator: ComparisonOperator;
+};
+
+// straightforward since they are represented in the same way that echarts represents its colors
+const getColorValuePairsForLTandLTE = (allThresholds: ThresholdRange[]) => {
+  const lineStylesLTandLTE: [number, string][] = [];
+
+  const lteThresholds = allThresholds.filter(
+    (tr) => tr.comparisonOperator === 'LT' || tr.comparisonOperator === 'LTE'
+  );
+
+  lteThresholds.forEach((thresholdRange) => {
+    const { end, color } = thresholdRange;
+    end && lineStylesLTandLTE.push([end, color]);
+  });
+
+  return lineStylesLTandLTE;
+};
+
+// more complicated since GT and GTE are colored in the opposite way that echarts represents its colors
```

Comment:
What's the value with saying it's more complicated when you can see it in the code. Could you instead explain why it's more complicated?